src/Summoner/License.hs

```diff
@@ -15,25 +18,47 @@ import qualified Data.Text as T
 -- License
 ----------------------------------------------------------------------------
 
-licenseNames :: [License]
-licenseNames = map fst githubLicenseQueryNames
-
-githubLicenseQueryNames :: [(License, Text)]
-githubLicenseQueryNames =
-    [ ("MIT",        "mit")
-    , ("BSD2",       "bsd-2-clause")
-    , ("BSD3",       "bsd-3-clause")
-    , ("GPL-2",      "gpl-2.0")
-    , ("GPL-3",      "gpl-3.0")
-    , ("LGPL-2.1",   "lgpl-2.1")
-    , ("LGPL-3",     "lgpl-3.0")
-    , ("AGPL-3",     "agpl-3.0")
-    , ("Apache-2.0", "apache-2.0")
-    , ("MPL-2.0",    "mpl-2.0")
-    ]
-
-newtype License = License { unLicense :: Text }
-    deriving (IsString, Eq, Ord, Show)
+data LicenseName = MIT
+             | BSD2
+             | BSD3
+             | GPL2
+             | GPL3
+             | LGPL21
+             | LGPL3
+             | AGPL3
+             | Apache20
+             | MPL20
+             deriving (Eq, Ord, Show, Enum, Bounded, Generic)
+
+githubLicenseQueryNames :: LicenseName -> Text
+githubLicenseQueryNames MIT      = "mit"
+githubLicenseQueryNames BSD2     = "bsd-2-clause"
+githubLicenseQueryNames BSD3     = "bsd-3-clause"
+githubLicenseQueryNames GPL2     = "gpl-2.0"
+githubLicenseQueryNames GPL3     = "gpl-3.0"
+githubLicenseQueryNames LGPL21   = "lgpl-2.1"
+githubLicenseQueryNames LGPL3    = "lgpl-3.0"
+githubLicenseQueryNames AGPL3    = "agpl-3.0"
+githubLicenseQueryNames Apache20 = "apache-2.0"
+githubLicenseQueryNames MPL20    = "mpl-2.0"
+
+showLicense :: LicenseName -> Text
+showLicense MIT      = "MIT"
+showLicense BSD2     = "BSD2"       
+showLicense BSD3     = "BSD3"       
+showLicense GPL2     = "GPL-2"      
+showLicense GPL3     = "GPL-3"      
+showLicense LGPL21   = "LGPL-2.1"   
+showLicense LGPL3    = "LGPL-3"     
+showLicense AGPL3    = "AGPL-3"     
+showLicense Apache20 = "Apache-2.0" 
+showLicense MPL20    = "MPL-2.0"
```

Comment:
It's better to name this function `cabalLicenseName`. It's main purpose is to use in `.cabal` file.